port/osx/omrintrospect.c

```diff
@@ -210,60 +243,48 @@ static int
 setupNativeThread(J9ThreadWalkState *state, thread_context *sigContext)
 {
 	PlatformWalkData *data = (PlatformWalkData *)state->platform_data;
-	int size = sizeof(thread_context);
 	int rc = 0;
 
-	if (size < sizeof(ucontext_t)) {
-		size = sizeof(ucontext_t);
-	}
-
 	/* Allocate the thread container. */
 	state->current_thread = (J9PlatformThread *)state->portLibrary->heap_allocate(state->portLibrary, state->heap, sizeof(J9PlatformThread));
 	if (NULL == state->current_thread) {
+		RECORD_ERROR(state, ALLOCATION_FAILURE, 2);
 		rc = -1;
 	}
 	if (0 == rc) {
 		memset(state->current_thread, 0, sizeof(J9PlatformThread));
-
-		/* Allocate space for the copy of the context. */
-		state->current_thread->context = (thread_context *)state->portLibrary->heap_allocate(state->portLibrary, state->heap, size);
-		if (NULL == state->current_thread->context) {
-			rc = -1;
-		}
 	}
 
 	if (0 == rc) {
-		state->current_thread->thread_id = data->threadList[data->threadIndex];
+		uint64_t tid = 0;
+		pthread_threadid_np(pthread_from_mach_thread_np(data->threadList[data->threadIndex]), &tid);
```

Comment:
At a number of places, function calls are being nested. Other examples:
```
pthread_kill(pthread_from_mach_thread_np ...
timedWait(timeout ...
```
Will there be gaps in error handling if one function passes a bad value to the other function?

Reply:
@EricYangIBM ^^^ Your thoughts?

Reply:
I fixed the first occurrence. For the `pthread_kill`, according to https://github.com/apple/darwin-libpthread/blob/main/src/pthread.c `pthread_kill` will return non-zero if `th` is null, which will skip the rest of the stack collection for that thread. For `timedWait`, the only error input (from `timeout`) is 0, which will be treated as a timeout.